File: euler2.cpp
//Copyright Jakob Sinclair 2015
#include <cmath>
#include <iostream>

int main() {
	int x = 1, y = 2, sum = 2;
	while((x < pow(10, 6) * 4) && (y < pow(10, 6) * 4)) {
		if(x < y) {
			x = x + y;
			if(x % 2 == 0) sum += x;
			std::cout << x << std::endl;
		} else {
			y = y + x;
			if(y % 2 == 0) sum += y;
			std::cout << y << std::endl;
		}
	}
	std::cout << sum << std::endl;
	return 0;
}

File: src/rust/euler2.rs
/*
The MIT License (MIT)

Copyright (c) 2016 Jakob Sinclair

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.
*/

fn main()
{
        let mut sum = 2;
        let mut x = 2;
        let mut y = 1;
        
        loop {
                if x < 4000000 && y < 4000000 {
                        if x < y {
                                x += y;
                                if x % 2 == 0 {
                                        sum += x;
                                }
                        } else {
                                y += x;
                                if y % 2 == 0 {
                                        sum += y;
                                }
                        }
                } else {
                        break;
                }
        }

        println!("Answer: {}", sum);
}
